src/paper/tasks.py

```diff
@@ -1269,45 +1308,97 @@ def pull_openalex_author_works(user_id, openalex_id):
     return True
 
 
-# Temporarily disabled as it was causing a lot of slowness in production
-# @periodic_task(
-#     # run at 4:30 PM UTC (8:30 AM PST)
-#     run_every=crontab(minute=30, hour=16), priority=3, queue=QUEUE_PULL_PAPERS
-# )
-def pull_new_openalex_works(page=0, retry=0):
+@periodic_task(
+    # run at 4:30 PM UTC (8:30 AM PST)
+    run_every=crontab(minute=30, hour=16), priority=3, queue=QUEUE_PULL_PAPERS
+)
+def pull_new_openalex_works(start_index=0, retry=0, paper_fetch_log_id=None):
+    """
+    Pull new works (papers) from OpenAlex.
+    
+    This looks complicated because we're trying to handle retries and logging.
+    But simply:
+    1. Get new works from OpenAlex in batches
+    2. Kick-off a task to create/update papers for each work
+    3. If we hit an error, retry the job from where we left off
+    4. Log the results
+    """
     if not PRODUCTION:
         return
 
     from paper.models import PaperFetchLog
 
-    sentry.log_info("Starting New OpenAlex pull")
+    fetch_since_date = datetime.now() - timedelta(days=1)
+    # if paper_fetch_log_id is provided, it means we're retrying
+    # otherwise we're starting a new pull
+    if paper_fetch_log_id is None:
+        start_date = datetime.now()
 
-    start_date = datetime.now()
+        # figure out when we should start fetching from.
+        # if we have an existing successful run, we start from the last successful run
+        try:
+            last_log = PaperFetchLog.objects.filter(
```

Comment:
Can we change name to `last_successful_run_log` or something like that. Kinda verbose but it does help when reading this.

Reply:
Yup I like that, it's complicated code so good to have verbosity.